src/utils/constants.js

```diff
@@ -69,6 +69,12 @@ StorageKeys.IS_READY_TO_PAY = 'isreadytopay';
  */
 StorageKeys.REDIRECT = 'subscribe.google.com:rk';
 
+/**
+ * Local storage key for should show autoprompt timestamps.
+ * const {string}
+ */
+StorageKeys.SHOULD_SHOW_AUTOPROMPT = 'shouldshowautopromptexp';
```

Comment:
Nit: This name seems more accurate to me, since the code is saving this event even when the prompt is suppressed
```suggestion
StorageKeys.SECOND_PROMPT_DELAY_COUNTER= 'secondpromptdelaycounter';
```

Reply:
> One thing to note here is, it's designed for new visitors (starting from 1st prompt), however there are users who have already visited and may have seen 1st prompt, but are put in the experiment as if they are first visitor (timestamps are collected from n-th prompt). We will need to handle the transitions between before/after this change if that could lead to an unexpected triggering behavior. Fortunately, it doesn't seem like a big issue for now as we're going to selectively run the experiment for one publisher who have only 2 types of prompt and the possible issues I can think of is, the experiment doesn't take effect if 1st prompt was already shown. That being said, the approach looks good to me. We will need to think about how we can distinguish it in the experiment data though.

Good call out, perhaps a solution is to add extra checks to verify a user is a "new visitor" (at least to our best ability with anonymous readers), such as checking no other prompts were previously dismissed.